libqtile/widget/net.py

```diff
@@ -29,43 +29,45 @@ class Net(base.ThreadedPollText):
     """Displays interface down and up speed"""
     orientations = base.ORIENTATION_HORIZONTAL
     defaults = [
-        ('interface', 'wlan0', 'The interface to monitor'),
+        ('interface', None, 'List of interfaces to monitor or None to displays all active NICs combined'),
         ('update_interval', 1, 'The update interval.'),
     ]
 
     def __init__(self, **config):
         base.ThreadedPollText.__init__(self, **config)
         self.add_defaults(Net.defaults)
-        self.interfaces = self.get_stats()
+        if not isinstance(self.interface, list):
+            if self.interface is None:
+                self.interface = "all"
+            self.interface = [self.interface]
+        self.stats = self.get_stats()
 
     def convert_b(self, b):
-        # Here we round to 1000 instead of 1024
-        # because of round things
-        letter = 'B'
-        # b is a float, so don't use integer division
-        if int(b / 1000) > 0:
-            b /= 1000.0
-            letter = 'k'
-        if int(b / 1000) > 0:
-            b /= 1000.0
-            letter = 'M'
-        if int(b / 1000) > 0:
-            b /= 1000.0
-            letter = 'G'
-        # I hope no one have more than 999 GB/s
-        return b, letter
 
-    def get_stats(self):
+        letters = ["b", "kb", "Mb", "Gb", "Tb", "Pb", "Eb", "Zb", "Yb"]
+        factor = 1000.0
+        i = 0
 
-        interfaces = {}
-        net = psutil.net_io_counters(pernic=True)
-        for iface in net:
-            name = iface
-            down = net[iface].bytes_recv
-            up = net[iface].bytes_sent
-            interfaces[name] = {'down': down, 'up': up}
+        while int(b / 1000) > 0:
+            b /= factor
+            i = i + 1
+
+        return b, letters[i]
 
-        return interfaces
+    def get_stats(self):
+        interfaces = {}
+        if self.interface == ["all"]:
+            net = psutil.net_io_counters(pernic=False)
+            interfaces["all"] = {'down': net[1], 'up': net[0]}
+            return interfaces
+        else:
+            net = psutil.net_io_counters(pernic=True)
+            for iface in net:
+                name = iface
+                down = net[iface].bytes_recv
+                up = net[iface].bytes_sent
+                interfaces[name] = {'down': down, 'up': up}
```

Comment:
You can just assign to `interfaces[iface]` and save the `name =` assignment above.

Reply:
Done!
I was pretty blind there...